packages/react-environment/src/api-providers/web3/web3-mixer-deposit.ts

```diff
@@ -45,18 +45,36 @@ export class Web3MixerDeposit extends MixerDeposit<WebbWeb3Provider, DepositPayl
   async generateNote(mixerAddress: string): Promise<DepositPayload> {
     const contract = await this.inner.getContractByAddress(mixerAddress);
     const mixerInfo = this.inner.getMixers().getMixerInfoByAddress(mixerAddress);
+
     if (!mixerInfo) {
       throw new Error(`mixer not found from storage`);
     }
+    const depositSizeBN = await contract.denomination;
+    const depositSize = Number.parseFloat(utils.fromWei(depositSizeBN.toString(), 'ether'));
+    const chainId = await this.inner.getChainId();
+    const deposit = createDeposit();
+    const secrets = deposit.preimage;
+    const noteInput: NoteGenInput = {
+      prefix: 'webb.mix',
+      chain: String(chainId),
+      amount: String(depositSize),
+      denomination: '18',
+      hashFunction: 'Poseidon5',
```

Comment:
Is this meant to be `Poseidon5x5`? Can we abstract hardcoded strings into a types/config file or file of enums?

Reply:
yeah, I Will change this to be just `Poseidon` since both the exponentiation and the width are now included with the Note